### DISCUSSION TITLE
What cuts and heuristics are used in HIGHS?

### DISCUSSION THREAD

**Discussion by maxBezel** [ERGO-Code/HiGHS#1846] [Q&A]
Hi! 
I'm currently using HIGHS to solve MILP problems, with the "node_limit" parameter set to 1 when calling from Python, or with "mip_max_nodes = 1" when using a version compiled from source code.

While exploring the source code, I came across the HighsCliqueTable class, which suggests that clique cuts are being utilized. However, I haven't been able to find any other cuts implementations, such as Gomory cuts or others.  Is it solving my problems with simplex method?

I would also love to know what heuristics are implemented in HIGHS and I would be grateful if you could explain the control flow of HIGHS on a high level.

Thanks in advance!

**Comment by jajhall** [comment]
> While exploring the source code, I came across the HighsCliqueTable class, which suggests that clique cuts are being utilized. However, I haven't been able to find any other cuts implementations, such as Gomory cuts or others. 

Yes, there are other cuts, such as Gomory cuts

> Is it solving my problems with simplex method?

For the nodes, the dual simplex solver is (naturally) being used. Currently the root nodes are being solved with simplex, but the default root node solution technique is expected to change to simplex/IPM according to which of the methods is expected to be faster

> I would also love to know what heuristics are implemented in HIGHS and I would be grateful if you could explain the control flow of HIGHS on a high level.

There many heuristics, but we don't have a document explaining these or "the control flow of HIGHS on a high level". The code is open-source...
